metacat-metadata-mysql/src/main/java/com/netflix/metacat/metadata/mysql/MySqlTagService.java

```diff
@@ -307,25 +325,31 @@ public void remove(final QualifiedName name, final Set<String> tags, final boole
         @Nullable final Set<String> excludeTags,
         @Nullable final String sourceName,
         @Nullable final String databaseName,
-        @Nullable final String tableName
+        @Nullable final String tableName,
+        @Nullable final QualifiedName.Type type
     ) {
         Set<String> includedNames = Sets.newHashSet();
         final Set<String> excludedNames = Sets.newHashSet();
-        final String wildCardName = QualifiedName.toWildCardString(sourceName, databaseName, tableName);
+        final String wildCardName =
+            MySqlServiceUtil.qualifiedNameToWildCardQueryString(sourceName, databaseName, tableName);
```

Comment:
Why did we replace the previous line with the new one?

Reply:
It has a similar issue, i.e. only limiting to table and can't return the database using search. From the API signature, it doesn't say the search only limits to tables and we should return the database if it has the tag.